% PLEASE USE THIS FILE AS A TEMPLATE
% Check file iosart2c.tex for more examples
%
% Journal:
%   Journal of Ambient Intelligence and Smart Environments (jaise)
%   Web Intelligence and Agent Systems: An International Journal (wias)
%   Semantic Web: Interoperability, Usability, Applicability (SW)
% IOS Press
% Latex 2e

% options: jaise|wias|sw
% add. options: [seceqn,secfloat,secthm,crcready,onecolumn]


%\documentclass{iosart2c}

\documentclass[sw]{iosart2c}
%\documentclass[wias]{iosart2c}
%\documentclass[jaise]{iosart2c}

\usepackage[T1]{fontenc}
\usepackage{times}%
\usepackage{natbib}% for bibliography sorting/compressing
%\usepackage{amsmath}
%\usepackage{endnotes}
\usepackage{graphics}

\usepackage{color}
\usepackage{url}
\usepackage{xspace}

%%%%%%%%%%% Put your definitions here

\newcommand{\myurl}[1]{\footnote{\url{#1}}}
\newcommand{\todo}[1]{\textbf{{\color{blue}$\Longrightarrow$ #1}}}

\newcommand{\via}{\emph{via}\xspace}
\newcommand{\perse}{\emph{per se}\xspace}
\newcommand{\ie}{\emph{i.e.}\xspace}
\newcommand{\eg}{\emph{e.g.}\xspace}
\newcommand{\etc}{\emph{etc.}}
\newcommand{\protege}{Prot\'eg\'e\xspace}
\newcommand{\con}[1]{\texttt{#1}\xspace}


%%%%%%%%%%% End of definitions

\pubyear{0000}
\volume{0}
\firstpage{1}
\lastpage{1}

\begin{document}

\begin{frontmatter}

%\pretitle{}
\title{From pure phytopathology to the Semantic Web world: the Plant-Pathogen Interactions Ontology (PPIO)}
\runningtitle{}
%\subtitle{}

%\review{}{}{}


% For one author:
%\author{\fnms{} \snm{}\thanks{}}
%\address{}
%\runningauthor{}

%Two or more authors:
\author[A]{\fnms{Alejandro} \snm{Rodr\'iguez Iglesias}\thanks{Corresponding author. Email:alejandroriglesias@gmail.com}},
\author[A]{\fnms{Mikel} \snm{Ega\~na Aranguren}}
\author[A]{\fnms{Alejandro} \snm{Rodr\'iguez Gonz\'alez}}
\author[A]{\fnms{Mark D.} \snm{Wilkinson}}
\runningauthor{}
\address[A]{Biological Informatics Group, Centre for Plant Biotechnology and Genomics (CBGP), Technical University of Madrid (UPM), Spain}
%\address[B]{}

\begin{abstract}
The scientific relevance of the studies based on plant-pathogenic bacteria interactions is undeniable. This is also important from the economic point of view of some plants species, and, of course, there is also a biodiversity component behind these studies. The implementation of the tools the Semantic Web offers into this area of knowledge has not been completely carried out. We present here the Plant-Pathogen Interactions Ontology (PPIO), whose axiomatic models allow the integration and inference of plant-pathogenic bacteria interactions datasets in an automated manner.
\end{abstract}

\begin{keyword}
 Plant pahogenic bacteria \sep Ontologies \sep Semantic Web \sep PPIO
\end{keyword}

\end{frontmatter}

%%%%%%%%%%% The article body starts:

\section{Introduction}\label{s1}

Plants can be susceptible to the attack of different pathogenic bacterial genera \cite{Mansfield}. If the different levels of the plant defense barriers are overcomed, the infection process can ultimately lead to the death of the plant. This phenomenon affects crop yield, and as a consequence there is also an impact on the economy around this species \cite{Montesinos}. Being able to explore new data sources data regarding plant-pathogenic bacteria interactions is an essential step, and will ultimately lead to a better preservation of worldwide crops. Hundreds of publications that focus their interest on unveiling the mechanisms of pathogenic bacteria interactions with their hosts reflect the biological significance of this area of research \cite{DeWit} \cite {Dodds}.

The field of plant-pathogenic bacteria is a good example of biodiversity richness \cite{Bull2010}. Traditionally, {\itshape Agrobacterium, Erwinia, Pseudomonas and Xanthomonas} were considered the four main plant pathogenic genera. With the rapid improvement of the research experimental approaches, the number of known plant pathogenic bacteria genera has increased up to 30. This has resulted in the generation of big amounts of unexplored biological data. The effectiveness of the use of Semantic technologies to manage with large sets of data has been already proved in other areas of life sciences, as discussed later. However, these tools that the Semantic Web offers have not been extensively applied to the knowledge domain of plant-pathogenic bacteria to date. Here we present the Plant Pathogen Interactions Ontology (PPIO), an ontology developed to collect data from the plant-pathogenic bacteria interactions domain. The main goal of PPIO is to serve as a reference for expert plant pathologists, providing the knowledge necessary to provide assistance in the interpretation of the phenotypic responses that result from these biological interactions.

%\subsection{}\label{s1.1}



\section{Modelling}

The initial data necessary to start building PPIO was collected manually by consulting a number of different expert web resources. The web page http://pseudomonas-syringae.org/ contained diverse state-of-art datasets related with various {\itshape Pseudomonas syringae} pathogenic strains. This page was a bridge to other web services where more datasets were collected \myurl{http://ncppb.fera.defra.gov.uk/}. After an initial collection of data was gathered and revised, the modelling of these datasets was performed using the ontology editor Prot\'eg\'e.

\subsection{Desing principles}

The main goal pursued during the modelling and designing process was to semantically capture as many biological data existing as possible, and special effort has been done in modelling the {\itshape disease triangle} \myurl{http://www.apsnet.org/edcenter/instcomm/TeachingArticles/Pages/DiseaseTriangle.aspx}. This term, one of the essentials milestones in plant pathology, asserts that three factors must be present for a disease to occur: a virulent pathogen, a susceptible host and a propitious environment. Two classes have been created to represent these three elements in a truthful and precise manner, the {\sf Environmental parameter} and the {\sf Organism} classes. This later class contains two subclasses that semantically describe both plant and pathogenic bacteria; these subclasses are linked to the {\sf NCBITaxon\_1} class, that incorporates both a bacterial and plant genera hierarchy with their corresponding taxa identifier. The connection between the three components of the disease triangle and the description of each one in the ontology is vital for accurately expressing this biological information into ontology language. To this end, terms such as {\sf 'Plant Pathogen'}, {\sf 'Host Plant'} or { \sf 'Resistant plant'} have been strongly axiomatically modelled to assure a trustworthy capture of the extracted biological data. By making a good use of the reasoning power this technology offers, the members of the {\sf Host Plant} and {\sf Resistant Plant} subclasses were inferred after the reasoning process took place (using the FaCT++ and HermiT 1.3.8 reasoners).

Physiological state of plants can be inferred visually by observing different phenotypes. Thus, significant endeavour has also been done in modelling plant phenotypic representations in PPIO. Therefore, a number of classes have been specifically created to meticulously represent plant phenotypic traits. Specially important are the {\sf Phenotype} and the {\sf 'Phenotypic process'} classes. These two classes semantically illustrate the output of the interaction between the host and the bacteria, which is ultimately represented as a resistance or a suscepibility phenotype expressed in the plant.
 
The {\sf Trait} class contains various physiological, biochemical and molecular plant traits. To be able to work with the classes that contain these traits, the Plant Trait Ontology \myurl{http://www.gramene.org} platform \cite{PTO} has been imported and integrated into PPIO, converting this ontology into an archetype of the Semantic Web technologies capabilities for data integration. The traits described in the different PTO classes can be affected if a bacterial attack takes place, and this has also been semantically illustrated in PPIO by axiomatically relating the {\sf Trait} class with both {\sf Phenotype} and {\sf 'Phenotypic process'} classes.


\section{Creation methodology}

\subsection{URI design}
The ontology URI (\url{http://purl.oclc.org/PPIO}) is HTTP resolvable and permanent (the PURL server redirects to our current server at \url{biordf.org}). The identifiers for entities (classes, individuals and object properties) are alphanumeric, with a URI of the type \url{http://purl.oclc.org/PPIO#PPIO_NNNNNNN}, and every entity has an informative \con{rdfs:label} annotation. Currently hash URIs are used due to the small size of the ontology, but since URIs are generated programmatically\myurl{https://github.com/wilkinsonlab/OWLNumericIDGenerator}, when the ontology grows into a Knowledge Base or Linked Data dataset (see section \ref{sec:discussion}) slash URIs can be generated.

\subsection{Ontology production}
The development of PPIO is automated as much as possible. Once the main structure is set, most of the remaining parts are produced programmatically using the Galaxy platform, a bioinformatics-oriented workflow environment \cite{galaxy}. By using Galaxy, the specific workflow we need is defined once and executed for each release; also, we can plug PPIO directly with other Bioinformatics tools. 

The workflow adds the necessary entities and axioms\footnote{The workflow can be reproduced at \url{http://biordf.org:8090/u/alejandroriglesias/w/ppio-taxa-punning}} (Figure \ref{fig:galaxy-workflow}):

\begin{enumerate}

\item The organism taxa hierarchy is produced by the tool NCBITaxonomy2OWL\myurl{https://github.com/wilkinsonlab/NCBITaxonomy2OWL}: it gets the user-defined taxa from the NCBI taxononomy through a BioPortal Web Service \cite{bioportal} and injects them in PPIO, reproducing the original taxonomical hierarchy (representing each rank-subrank as a simple subsumption relation \cite{taxa_ismb_2008}) and adding each taxon with a resolvable OntoBee\myurl{http://www.ontobee.org/} URI.

\item Since pathogens in PPIO are modelled as individuals, they cannot be directly related with class hierarchies like the NCBI taxonomy and the symptoms hierarchy. Therefore, PPIO exploits OWL punning\myurl{http://www.w3.org/TR/owl2-new-features/Punning} and an individual with the same URI as each type class is generated programmatically (see bellow) for those hierarchies: the linking of pathogens to those hierarchies (\eg~\con{NCBITaxon\_552 types Erwinia amylovora}, \con{NCBITaxon\_552 causes symptom Canker}, \con{NCBITaxon\_552 causes symptom Blight}) is done manually. This is achieved by defining two Ontology Pre Procesor Language (OPPL)\myurl{http://oppl.sf.net} scripts and executing them via OPPL-Galaxy \cite{OPPL-Galaxy-JBMS}\footnote{Since we are using OPPL, any complex axiomisation -not only puning- can be defined once and automatically applied -expanded to different parts of the ontology- every time the workflow is executed.}:

{\small 
\begin{verbatim}
?x:CLASS,
?y:INDIVIDUAL = create(?x.RENDERING)
SELECT ?x SubClassOf NCBITaxon_1
WHERE ?x != Nothing, ?x != Thing
BEGIN
ADD ?y Type ?x
END;
\end{verbatim}
}

{\small 
\begin{verbatim}
?x:CLASS,
?y:INDIVIDUAL = create(?x.RENDERING)
SELECT ?x SubClassOf PPIO_0000069
WHERE ?x != Nothing, ?x != Thing
BEGIN
ADD ?y Type ?x
END;
\end{verbatim}
}


\end{enumerate}

\begin{figure*}
%\includegraphics
%[width=.1\textwidth]
%{galaxy-workflow}
\caption{Galaxy workflow for producing a release of PPIO. In the first step, NCBITaxonomy2OWL is executed; it gets the ontology and a flat file containing the NCBI taxonomy IDs, and it adds them to the ontology. Then two OPPL scripts are executed against the resulting ontology, adding axioms and entities to create }\label{fig:galaxy-workflow}
\end{figure*}

\section{Discussion}\label{sec:discussion}

Semantic-oriented platforms like the OBO foundry \cite{Smith},which includes Gene Ontology (GO) \cite{Gene}, Bio2RDF \cite{RDF} or the W3C Semantic Web for Health Care and Life Sciences Interest Group\myurl{http://www.w3.org/blog/hcls/} are excellent paradigms of the success in using semantics to assist in the integration of automated data. Nevertheless, taking a look at the fields of plant biotechnology and phytopathology, it is surprising to notice that these technologies have been applied to a limited number of domain resources. There are some precedents prior to the ontology discussed here, such as the Plant Ontology\myurl{tp://www.plantontology.org/}, which describes plant anatomy, morphology and developmental stages \cite{PO}. The Plant Disease Ontology (IDOPlant) \cite{Walls} \cite{IDO}, on the other hand, is focused on generically describing plant infectious diseases. Finally, the GO extension for description of the Type III Effectors \cite{Lindeberg} is maybe the ontological contribution in the plant pathology and microbiology area more related to PPIO.201

In a comparison between the IDOPlant and PPIO, a more generalistic approach of data modelling can be appreciated in the case of the first platform, which describes plant infectious diseases caused by either biotic or abiotic agents. The ontology reported here pursues a knowledge capture strategy focused on data that concerns plant-pathogenic bacteria interactions, in order to be able to properly represent this area of knowledge. On the other hand, although the GO extension for the type III effectors is built for capturing processes at the host-pathogen level, effector proteins data capture is emphasized. So, the principal reason for building this ontology is to ensure the semantic description of a domain not entirely represented by other resources. Of course, PPIO has been designed to complement these previous ontologies by introducing accurate biological information. It is our objective to continuing the process of data integration not only importing related ontologies, but other resources such as the Darwin Core glossary of terms (DwC) \cite{Wieczorek2013}. The final goal of this initiative is to use this platform, combined with others, as a diagnosis/prevention/alert system. PPIO will make it possible for users to pose, and answer, questions like the following ones and obtain a meaningful answer:

\begin{enumerate}
\item Is {\itshape Solanum lycopersicum} susceptible to the attack of {\itshape Pseudomonas syringae} pv. {\itshape tomato} DC3000?
\item Does a high humidity favours the development of {\itshape Pectobacterium carotovorum} subsp. {\itshape carotovorum}?
\item What is the phenotype of the disease produced by {\itshape Dickeya dadantii} in {\itshape Solanum tuberosum}?
\item What is the host range of the pathogen {\itshape Pseudomonas marginalis} pv. {\itshape marginalis}?

\end{enumerate}

Knowledge acquisition is based on the process of transcripting the knowledge from unstructured sources into a format that is machine-readable and useful. This approach can report a great benefit if big amounts of datasets are required in order to populate ontologies. One of the main ideas behind this is that the participation of field experts should ensure the fiability of the data content captured. At the moment, a knowledge capture project is now being developed in our laboratory, and it is our thought that it will aid in the process of collecting data that will ultimately populate PPIO with trustworthy and scientifically relevant information. All these steps will help to our objective of converting PPIO into a basic and essential bioinformatic tool for scientific community in the area plant pathogens.

\section*{Acknowledgements}
Alejandro Rodr\'iguez Iglesias and Alejandro Rodr\'iguez Gonz\'alez are funded by the Isaac Peral Programme. Mark D. Wilkinson and Mikel Ega\~na Aranguren are funded by the Marie Curie-COFUND Programme (FP7) of the EU.



%\begin{figure}[t]
%\includegraphics{}
%\caption{Figure caption.}\label{f1}
%\end{figure}

%\begin{table*}
%\caption{} \label{t1}
%\begin{tabular}{lll}
%\hline
%&&\\
%&&\\
%\hline
%\end{tabular}
%\end{table*}


%%%%%%%%%%% The bibliography starts:
%\begin{thebibliography}{9}

%\bibitem{r1}

%\bibitem{r2}

%\end{thebibliography}

\bibliographystyle{abbrv}
  \bibliography{swj_ppio}

\end{document}
