% PLEASE USE THIS FILE AS A TEMPLATE
% Check file iosart2c.tex for more examples
%
% Journal:
%   Journal of Ambient Intelligence and Smart Environments (jaise)
%   Web Intelligence and Agent Systems: An International Journal (wias)
%   Semantic Web: Interoperability, Usability, Applicability (SW)
% IOS Press
% Latex 2e

% options: jaise|wias|sw
% add. options: [seceqn,secfloat,secthm,crcready,onecolumn]


%\documentclass{iosart2c}

\documentclass[sw]{iosart2c}
%\documentclass[wias]{iosart2c}
%\documentclass[jaise]{iosart2c}

\usepackage[T1]{fontenc}
\usepackage{times}%
\usepackage{natbib}% for bibliography sorting/compressing
%\usepackage{amsmath}
%\usepackage{endnotes}
%\usepackage{graphics}

\usepackage{url}

%%%%%%%%%%% Put your definitions here

\newcommand{\myurl}[1]{\footnote{\url{#1}}}


%%%%%%%%%%% End of definitions

\pubyear{0000}
\volume{0}
\firstpage{1}
\lastpage{1}

\begin{document}

\begin{frontmatter}

%\pretitle{}
\title{Plant-Pathogen Interactions Ontology (PPIO)}
\runningtitle{}
%\subtitle{}

%\review{}{}{}


% For one author:
%\author{\fnms{} \snm{}\thanks{}}
%\address{}
%\runningauthor{}

%Two or more authors:
\author[A]{\fnms{Alejandro} \snm{Rodr\'iguez Iglesias}\thanks{Corresponding author. Email:alejandroriglesias@gmail.com}},
\author[A]{\fnms{Mikel} \snm{Ega\~na Aranguren}}
\author[A]{\fnms{Alejandro} \snm{Rodr\'iguez Gonz\'alez}}
\author[A]{\fnms{Mark D.} \snm{Wilkinson}}
\runningauthor{}
\address[A]{Biological Informatics Group, Centre for Plant Biotechnology and Genomics (CBGP), Technical University of Madrid (UPM), Spain}
%\address[B]{}

\begin{abstract}
Plant-pathogen interactions are an important knowledge domain within plant biology and biotechnology, both scientifically and in economic terms. Unlike other knowledge domains within   life sciences, however, semantic technologies have not been used extensively to codify it; therefore, there is a lack of axiomatic models amenable to automated integration and inference. We present the Plant-Pathogen Interactions Ontology (PPIO), a first step towards the axiomatization of plant-pathogen interactions knowledge. PPIO encourages consistent annotation and supports both query and inference.
\end{abstract}

\begin{keyword}
 Plant pahogenic bacteria \sep Ontologies \sep Semantic Web 
\end{keyword}

\end{frontmatter}

%%%%%%%%%%% The article body starts:

\section{Introduction}\label{s1}

Many bacterial genera are responsible for causing diseases in a wide range of plant species, and if the  the different levels of the host defense barriers are overcomed by the pathogen, the infection process can ultimately lead to the death of the plant. This phenomenon can affect crop yield, which automatically will translate into considerable economic loss. On the other hand, in the context of challenges such as feeding a growing world population, and the trend of emerging economies to consume more meat, where animal feed increases the consumption per capita of available food-crops, it is crucial that we identify ways to increase the productivity of crop fields. Therefore, being able to accurately record, explore, and query biological data regarding plant-pathogen interactions is a necessary step towards the improvement and protection of these important plant species [5, 22].  This issue continues being subject of intensive research worldwide. This is documented in hundreds of articles that focus on the biological consequences and the mechanisms of pathogenic bacteria interactions with their hosts[30]. With modern high-throughput technologies, the amount and complexity of data and knowledge extracted from these studies is increasingly large [12, 20] and new information is unveiled every day.  This is happening in parallel with development of new experimental research techniques, as well as the application of bioinformatics tools in this field[6, 19].

%\subsection{}\label{s1.1}



\section{Modelling}
\subsection{Desing principles}



\section{Creation methodology}

Numerical IDs + label.

The development of PPIO is automated as much as possible. Once the basic structure is set, some parts of the ontology are produced programmatically by introducing the ontology in a tailored Galaxy \cite{galaxy} workflow  (picture of workflow). 

OWL Puning\myurl{http://www.w3.org/TR/owl2-new-features/Punning} is used in the X hierarchy of PPIO to represent X classes both as OWL classes and individuals, in order to link pathogens, which are individuals, with symptons?. This is achieved by defining an Ontology Pre Procesoor Language (OPPL)\myurl{http://oppl.sf.net} script (picture) and executing it via OPPL-Galaxy \cite{OPPL-Galaxy-JBMS}.

The organism taxa hierarchy is produced by the Galaxy tool NCBITaxonomy2OWL: NCBITaxonomy2OWL gets the user-defined taxa from the NCBO web service (URL) and injects them in the ontology, respecting the origincal hierarchy and adding each taxa with an ontobee (REF) URI (eg) (ref github repo)

By using Galaxy, the specifi workflow we need is defined once and we can execute it at will, each time a new release is set, or also if new tools are needed. The galaxy workflow can be reproduced at biordf.org:8090 with any ontology and OPPL script.

TODO: we need to be able to resolve not only the whole ontology (oclc.purl.org/PPIO), but specific entities (oclc.purl.org/PPIO000023). How? stardog linked data? but then, the redirection is always done with oclc.purl.org!



\section{Discussion (comparison with other ontologies on the same topic, pointers to existing applications or use-case experiments)}

\section*{Acknowledgements}

Mikel Ega\~na Aranguren is funded by the Marie Curie-COFUND Programme (FP7) of the EU.



%\begin{figure}[t]
%\includegraphics{}
%\caption{Figure caption.}\label{f1}
%\end{figure}

%\begin{table*}
%\caption{} \label{t1}
%\begin{tabular}{lll}
%\hline
%&&\\
%&&\\
%\hline
%\end{tabular}
%\end{table*}


%%%%%%%%%%% The bibliography starts:
%\begin{thebibliography}{9}

%\bibitem{r1}

%\bibitem{r2}

%\end{thebibliography}

\bibliographystyle{abbrv}
  \bibliography{swj_ppio}

\end{document}
