% PLEASE USE THIS FILE AS A TEMPLATE
% Check file iosart2c.tex for more examples
%
% Journal:
%   Journal of Ambient Intelligence and Smart Environments (jaise)
%   Web Intelligence and Agent Systems: An International Journal (wias)
%   Semantic Web: Interoperability, Usability, Applicability (SW)
% IOS Press
% Latex 2e

% options: jaise|wias|sw
% add. options: [seceqn,secfloat,secthm,crcready,onecolumn]


%\documentclass{iosart2c}

\documentclass[sw]{iosart2c}
%\documentclass[wias]{iosart2c}
%\documentclass[jaise]{iosart2c}

\usepackage[T1]{fontenc}
\usepackage{times}%
\usepackage{natbib}% for bibliography sorting/compressing
%\usepackage{amsmath}
%\usepackage{endnotes}
\usepackage{graphics}

\usepackage{color}
\usepackage{url}
\usepackage{xspace}

%%%%%%%%%%% Put your definitions here

\newcommand{\myurl}[1]{\footnote{\url{#1}}}
\newcommand{\todo}[1]{\textbf{{\color{blue}$\Longrightarrow$ #1}}}

\newcommand{\via}{\emph{via}\xspace}
\newcommand{\perse}{\emph{per se}\xspace}
\newcommand{\ie}{\emph{i.e.}\xspace}
\newcommand{\eg}{\emph{e.g.}\xspace}
\newcommand{\etc}{\emph{etc.}}
\newcommand{\protege}{Prot\'eg\'e\xspace}
\newcommand{\con}[1]{\texttt{#1}\xspace}


%%%%%%%%%%% End of definitions

\pubyear{0000}
\volume{0}
\firstpage{1}
\lastpage{1}

\begin{document}

\begin{frontmatter}

%\pretitle{}
\title{Plant-Pathogen Interactions Ontology (PPIO)}
\runningtitle{}
%\subtitle{}

%\review{}{}{}


% For one author:
%\author{\fnms{} \snm{}\thanks{}}
%\address{}
%\runningauthor{}

%Two or more authors:
\author[A]{\fnms{Alejandro} \snm{Rodr\'iguez Iglesias}\thanks{Corresponding author. Email:alejandroriglesias@gmail.com}},
\author[A]{\fnms{Mikel} \snm{Ega\~na Aranguren}}
\author[A]{\fnms{Alejandro} \snm{Rodr\'iguez Gonz\'alez}}
\author[A]{\fnms{Mark D.} \snm{Wilkinson}}
\runningauthor{}
\address[A]{Biological Informatics Group, Centre for Plant Biotechnology and Genomics (CBGP), Technical University of Madrid (UPM), Spain}
%\address[B]{}

\begin{abstract}
Plant-pathogen interactions are an important knowledge domain within plant biology and biotechnology, both scientifically and in economic terms. Unlike other knowledge domains within   life sciences, however, semantic technologies have not been used extensively to codify it; therefore, there is a lack of axiomatic models amenable to automated integration and inference. We present the Plant-Pathogen Interactions Ontology (PPIO), a first step towards the axiomatization of plant-pathogen interactions knowledge. PPIO encourages consistent annotation and supports both query and inference.
\end{abstract}

\begin{keyword}
 Plant pahogenic bacteria \sep Ontologies \sep Semantic Web \sep PPIO
\end{keyword}

\end{frontmatter}

%%%%%%%%%%% The article body starts:

\section{Introduction}\label{s1}

Plants can be susceptible to the attack of many different pathogenic bacterial genera \cite{Mansfield}, and if the different levels of the host plant defense barriers are overcomed by the pathogen, the infection process can ultimately lead to the death of the plant. This phenomenon affects crop yield, which automatically will translate into considerable economic loss \cite{Montesinos}. Therefore, being able to accurately explore and query biological data regarding plant-pathogenic bacteria interactions is a necessary step towards the improvement and protection of worldwide crops. This issue continues being subject of intensive research worldwide. Hundreds of articles that focus on the mechanisms of pathogenic bacteria interactions with their hosts reflect the biological significance of this area of research \cite{DeWit} \cite {Dodds}. 

The area of plant-pathogenic bacteria is an illustrious example of biodiversity richness \cite{Bull2010}. Traditionally, {\itshape Agrobacterium, Erwinia, Pseudomonas and Xanthomonas} were considered the four main plant pathogenic genera some years ago. Nowadays, and thanks to the development of new research techniques, the number of known plant pathogenic bacteria genera has increased up to 30. One of the direct consequences from these discoverys is the generation of vast amounts of unexplored data. The effectiveness of the use of Semantic technologies in other study areas of life sciences has been already proved. However, these tools have not been extensively applied to the knowledge domain of plant-pathogenic bacteria to date. Here we present the Plant Pathogen Interactions Ontology (PPIO), an ontology developed to semantically collect data in the context of the plant-pathogenic bacteria interactions. The main goal of PPIO is to ultimately serve as a reference ontology in this area, providing the knowledge necessary to assist in the interpretation of the phenotypic responses that result from the interaction established during the infection of the pathogenic bacteria and the potential host plants.

%\subsection{}\label{s1.1}



\section{Modelling}

The initial data necessary to start building our ontology was collected manually by consulting a number of different of web resources. The web page http://pseudomonas-syringae.org/ was the starting point; it contains diverse state-of-art datasets related with various {\itshape Pseudomonas syringae} pathogenic strains. This web page was a bridge to other web resources where more datasets were collected. After an initial collection of data was gathered, the modelling of these datasets was performed using the ontology editor Prot\'eg\'e (version). 

\subsection{Desing principles}

The main goal pursued during the modelling and designing process was to semantically capture as many biological data existing as possible, and special effort is being done in accurately modelling the {\itshape disease triangle} \myurl{http://www.apsnet.org/edcenter/instcomm/TeachingArticles/Pages/DiseaseTriangle.aspx} . This term is one of the most essential milestones in the plant pathology ambit, and it principal statement asserts that three factors must be present for a disease to occur: a virulent pathogen, a susceptible host and a propitious environment for disease progression. Two classes have been created to represent these three elements, the {\sf Environmental parameter} and the {\sf Organism} classes. This later class contains two subclasses that semantically capture both plant and bacterial genera. The connection between these three components and the precise description of each one in the ontology is vital for accurately expressing this biological information into ontology language. To this end, terms such as {\sf 'Plant Pathogen'} or { \sf 'Resistant plant'} are being strongly axiomatically modelled to assure a trustworthy capture of the extracted biological data.

Physiological state of plants can be inferred visually by observing different phenotypes. Thus, significant endeavour is also being done in modelling plant phenotypic representations in PPIO. To this end, a number of classes have been specifically created to represent plant phenotypic traits in a precise and pertinent manner. Specially important are the { \sf Phenotpype} and the {\sf 'Phenotypic process'} classes. These two classes semantically illustrate the output of the interaction between the host and the bacteria, which is ultimately represented as a resistance or a suscepibility phenotype expressed in the plant. The {\sf Trait} class contains physiological, biochemical and molecular plant traits imported from the Plant Trait Ontology \myurl{http://www.gramene.org}. These traits can be affected if a bacterial attack takes places, and this has also been semantically expressed in PPIO by axiomatically relating this class with both {\sf Phenotype} and {\sf 'Phentypic process'} classes. 


\section{Creation methodology}

\subsection{URI design}
The ontology URI (\url{http://purl.oclc.org/PPIO}) is HTTP resolvable and permanent (the PURL server redirects to our current server at \url{biordf.org}). The identifiers for entities (classes, individuals and object properties) are alphanumeric, with a URI of the type \url{http://purl.oclc.org/PPIO#PPIO_NNNNNNN}, and every entity has an informative \con{rdfs:label} annotation. Currently hash URIs are used due to the small size of the ontology, but since URIs are generated programmatically\myurl{https://github.com/wilkinsonlab/OWLNumericIDGenerator}, when the ontology grows into a Knowledge Base or Linked Data dataset (see section \ref{sec:discussion}) slash URIs can be generated.

\subsection{Ontology production}
The development of PPIO is automated as much as possible. Once the main structure is set, most of the remaining parts are produced programmatically using the Galaxy platform, a bioinformatics-oriented workflow environment \cite{galaxy}. By using Galaxy, the specific workflow we need is defined once and executed for each release; also, we can plug PPIO directly with other Bioinformatics tools. 

The workflow adds the necessary entities and axioms\footnote{The workflow can be reproduced at \url{http://biordf.org:8090/u/alejandroriglesias/w/ppio-taxa-punning}} (Figure \ref{fig:galaxy-workflow}):

\begin{enumerate}

\item The organism taxa hierarchy is produced by the tool NCBITaxonomy2OWL\myurl{https://github.com/wilkinsonlab/NCBITaxonomy2OWL}: it gets the user-defined taxa from the NCBI taxononomy through a BioPortal Web Service \cite{bioportal} and injects them in PPIO, reproducing the original taxonomical hierarchy (representing each rank-subrank as a simple subsumption relation \cite{taxa_ismb_2008}) and adding each taxon with a resolvable OntoBee\myurl{http://www.ontobee.org/} URI.

\item Since pathogens in PPIO are modelled as individuals, they cannot be directly related with class hierarchies like the NCBI taxonomy and the symptoms hierarchy. Therefore, PPIO exploits OWL punning\myurl{http://www.w3.org/TR/owl2-new-features/Punning} and an individual with the same URI as each type class is generated programmatically (see bellow) for those hierarchies: the linking of pathogens to those hierarchies (\eg~\con{NCBITaxon\_552 types Erwinia amylovora}, \con{NCBITaxon\_552 causes Canker}, \con{NCBITaxon\_552 causes Blight}) is done manually. This is achieved by defining two Ontology Pre Procesor Language (OPPL)\myurl{http://oppl.sf.net} scripts and executing them via OPPL-Galaxy \cite{OPPL-Galaxy-JBMS}\footnote{Since we are using OPPL, any complex axiomisation -not only puning- can be defined once and automatically applied -expanded to different parts of the ontology- every time the workflow is executed.}:

{\small 
\begin{verbatim}
?x:CLASS,
?y:INDIVIDUAL = create(?x.RENDERING)
SELECT ?x SubClassOf NCBITaxon_1
WHERE ?x != Nothing, ?x != Thing
BEGIN
ADD ?y Type ?x
END;
\end{verbatim}
}

{\small 
\begin{verbatim}
?x:CLASS,
?y:INDIVIDUAL = create(?x.RENDERING)
SELECT ?x SubClassOf PPIO_0000069
WHERE ?x != Nothing, ?x != Thing
BEGIN
ADD ?y Type ?x
END;
\end{verbatim}
}


\end{enumerate}

\begin{figure*}
% \includegraphics
% [width=.1\textwidth]
% {galaxy-workflow}
\caption{Galaxy workflow for producing a release of PPIO. In the first step, NCBITaxonomy2OWL is executed; it gets the ontology and a flat file containing the NCBI taxonomy IDs, and it adds them to the ontology. Then two OPPL scripts are executed against the resulting ontology, adding axioms and entities to create }\label{fig:galaxy-workflow}
\end{figure*}

\section{Discussion}\label{sec:discussion}

Certain semantics-oriented platforms are particularly noteworthy as a result of their success in using semantics to aid in automated data  integration from non-collaborating resources, like the OBO foundry \cite{Smith},which includes Gene Ontology (GO) \cite{Gene}, Bio2RDF \cite{RDF} and the W3C Semantic  Web  for Health  Care  and  Life Sciences Interest Group\myurl{http://www.w3.org/blog/hcls/}. In the context of plant biotechnology and phytopathology, however, semantic technologies have been applied to only a limited number of domain resources. For example, the Plant Ontology Consortium developed the Plant Ontology\myurl{tp://www.plantontology.org/} platform to describe the anatomy, morphology and developmental stages of plants \cite{PO}. The Plant Trait Ontology \cite{PTO}, related to the Gramene project, defines a vocabulary for describing the specific appearance or qualities of various plant anatomical structures. The Plant Disease Ontology (IDOPlant) \cite{Walls}, was modeled after the human Infectious Diseases Ontology (IDO) \cite{IDO} to serve as a reference ontology that contained and covered any plant infectious disease. Apart from IDOPlant, there is another contribution directly related to the plant-pathogenic bacteria area, and is the GO extension for description of the Type III Effectors \cite{Lindeberg}, which is part of the Plant-Associated Microbe Gene Ontology project (PAMGO). 


Although the existence of ontologies related with plants and plant pathology has already been reported \cite{PTO} \cite{Lindeberg} \cite{Walls}, the justification for building this ontology lies in its coverage of a domain not fully represented by other resources. In a comparison between the IDOPlant and PPIO, a more generalistic approach of data modelling can be seen in the case of the first platform, which semantically describes plant infectious diseases caused by either biotic or abiotic agents . The ontology reported here pursues a knowledge capture strategy specifically focused on data that concerns plant-pathogenic bacteria interactions. On the other hand, although the GO extension for the type III effectors is built for capturing processes in the host-pathogen level, effector proteins data capture is emphasized. Of course, PPIO has been developed in order to be used with these ontological resources. In this aspect, PPIO complements these previous ontologies by introducing accurate and concrete biological information concerning plant pathogenic bacteria and the interactions with plants. The final goal of this initiative is to use this platform, combined with others, as a diagnosis/prevention/alert system. PPIO will make it possible for users to pose, and answer, questions like the following ones and obtain a meaningful answer:

\begin{enumerate}
\item Is tomato plant susceptible to the attack of {\itshape Pseudomonas syringae} pv. {\itshape tomato} DC3000?
\item Does a high humidity and and low temperature rates favour the development of {\itshape Pectobacterium carotovorum} subsp. {\itshape carotovorum}?
\item What is the phenotype of the disease produced by {\itshape Dickeya dadantii} in {\itshape Solanum tuberosum}?

\end{enumerate}

Knowledge acquisition is based on the process of transcripting the knowledge from unstructured sources into a format that is machine-readable and useful. This approach can report a great benefit if big amounts of datasets are required in order to populate ontologies. One of the main ideas behind this that the participation of field experts should ensure the fiability of the data content captured. A knowledge capture project is now being developed in our laboratory, and it is our thought that it will aid in the process of collecting data that will ultimately populate PPIO. 

\section*{Acknowledgements}
Alejandro Rodr\'iguez Iglesias and Alejandro Rodr\'iguez Gonz\'alez are funded by the Isaac Peral Programme. Mark D. Wilkinson and Mikel Ega\~na Aranguren are funded by the Marie Curie-COFUND Programme (FP7) of the EU.



%\begin{figure}[t]
%\includegraphics{}
%\caption{Figure caption.}\label{f1}
%\end{figure}

%\begin{table*}
%\caption{} \label{t1}
%\begin{tabular}{lll}
%\hline
%&&\\
%&&\\
%\hline
%\end{tabular}
%\end{table*}


%%%%%%%%%%% The bibliography starts:
%\begin{thebibliography}{9}

%\bibitem{r1}

%\bibitem{r2}

%\end{thebibliography}

\bibliographystyle{abbrv}
  \bibliography{swj_ppio}

\end{document}
